exchange/bidder_test.go

```diff
@@ -3056,3 +3056,139 @@ func TestGetBidType(t *testing.T) {
 		})
 	}
 }
+
+func TestSeatNonBid(t *testing.T) {
+	type args struct {
+		BidRequest     *openrtb2.BidRequest
+		Seat           string
+		BidderResponse func() (*http.Response, error)
+	}
+	type expect struct {
+		// seatBids    []*entities.PbsOrtbSeatBid
+		seatNonBids *openrtb_ext.SeatNonBid
+		// errors      []error
+	}
+	testCases := []struct {
+		name   string
+		args   args
+		expect expect
+	}{
+		{
+			name: "101_timeout",
+			args: args{
+				Seat: "someseat",
+				BidRequest: &openrtb2.BidRequest{
+					Imp: []openrtb2.Imp{{ID: "1234"}},
+				},
+				BidderResponse: func() (*http.Response, error) { return nil, context.DeadlineExceeded },
+			},
+			expect: expect{
+				seatNonBids: &openrtb_ext.SeatNonBid{
+					Seat: "someseat",
+					NonBid: []openrtb_ext.NonBid{{
+						ImpId:      "1234",
+						StatusCode: int(ErrorTimeout),
+						Error:      context.DeadlineExceeded.Error(),
+						Ext:        openrtb_ext.NonBidExt{Prebid: openrtb_ext.ExtResponseNonBidPrebid{Bid: openrtb_ext.NonBidObject{Price: 0}}},
+					}}},
+				// errors:   []error{&errortypes.Timeout{Message: context.DeadlineExceeded.Error()}},
+				// seatBids: []*entities.PbsOrtbSeatBid{{Bids: []*entities.PbsOrtbBid{}, Currency: "USD", Seat: "someseat", HttpCalls: []*openrtb_ext.ExtHttpCall{}}},
+			},
+		}, /* {
+			name: "103_bidder_unreachable",
```

Comment:
Why is this test commented out?

Reply:
I was trying to make core unit test implementation reusable for future error code. will revert it for now